File: CPP/SSD.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

// Kelas Ssd yang merupakan turunan dari kelas Komponen
class Ssd : public Komponen {
    private:
        int kapasitasGB;
        string tipeDrive;
    
    public:
        Ssd() {}
    
          // Konstruktor dengan parameter kapasitasGB, tipeDrive, merk, dan nama
        Ssd(int kapasitasGB, string tipeDrive, string merk, string nama) 
            : Komponen(merk, nama) {
            this->kapasitasGB = kapasitasGB;
            this->tipeDrive = tipeDrive;
        }
    
        //setter
        void setKapasitasGB(int kapasitasGB) {
            this->kapasitasGB = kapasitasGB;
        }
    
        void setTipeDrive(string tipeDrive) {
            this->tipeDrive = tipeDrive;
        }
    
        //getter
        int getKapasitasGB() {
            return kapasitasGB;
        }
    
        string getTipeDrive() {
            return tipeDrive;
        }
    
    ~Ssd() 
    {

    }
};

File: python/Ssd.py
from Komponen import Komponen

class Ssd(Komponen):
    #atribut private
    __kapasitas_gb = ""
    __tipe_drive = ""

    #konstruktor
    def __init__(self, kapasitas_gb, tipe_drive, merk, nama):
        super().__init__(merk, nama)
        self._kapasitas_gb = kapasitas_gb
        self._tipe_drive = tipe_drive

    #setter
    def set_kapasitas_gb(self, kapasitas_gb):
        self._kapasitas_gb = kapasitas_gb

    def set_tipe_drive(self, tipe_drive):
        self._tipe_drive = tipe_drive

    #getter
    def get_kapasitas_gb(self):
        return self._kapasitas_gb

    def get_tipe_drive(self):
        return self._tipe_drive